### ISSUE TITLE
IPsec enable failed on EKS: unable to replace state: invalid argument

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#18856]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I'm failing to install Cilium with IPsec enabled on an EKS cluster. The installation of xfrm states apparently fails with `invalid argument` (see below).

### Cilium Version

quay.io/cilium/cilium:v1.11.1@sha256:251ff274acf22fd2067b29a31e9fda94253d2961c061577203621583d7e85bd2

### Kernel Version

5.4.176-91.338.amzn2.x86_64

### Kubernetes Version

v1.20.11-eks-f17b81

### Sysdump

[cilium-sysdump-20220218-132258.zip](https://github.com/cilium/cilium/files/8097075/cilium-sysdump-20220218-132258.zip)

### Relevant log output

```shell
level=error msg="IPsec enable failed" error="unable to replace remote state: invalid argument" local-ip=0.0.0.0/0 reason="CNI Out IPv4" remote-ip=192.168.0.0/16 spi=0 subsys=linux-datapath
level=error msg="IPsec enable failed" error="unable to replace local state: invalid argument" local-ip=192.168.147.13/19 reason="CNI In IPv4" remote-ip=0.0.0.0/0 spi=0 subsys=linux-datapath
```

### Reproduction steps
```
export COST_REDUCTION=(--nodes 2 --instance-types "t3.medium,t3a.medium" --node-volume-type gp3 --node-volume-size 10 --managed --spot)
eksctl create cluster --name "${NAME}" --region "${REGION}" --tags "${TAGS}" --without-nodegroup
./cilium install --cluster-name "${NAME}" --wait=false --rollback=false --encryption=ipsec
eksctl create nodegroup --cluster "${NAME}" --region "${REGION}" --node-private-networking "${COST_REDUCTION[@]}"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pchaigno** [comment]
It's also broken on k8s 1.18 and 1.19.

**Comment by tklauser** [comment]
This is likely the same failure as #18500. The fix #18506 was backported to the `v1.11` branch, but `v1.11.1` (the version reported in this issue) does not contain that fix yet.

**Comment by pchaigno** [comment]
Thanks Tobias! I can confirm the issue is gone if I use the `latest` images.
